keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/SumVertex.java

```diff
@@ -32,7 +38,18 @@ protected DualNumber dualOp(DualNumber dualNumber) {
 
     @Override
     public Map<Vertex, PartialDerivatives> reverseModeAutoDifferentiation(PartialDerivatives derivativeOfOutputsWithRespectToSelf) {
-        //TODO
-        throw new UnsupportedOperationException("Reverse mode autodiff not supported for this operation");
+
+        Map<VertexId, DoubleTensor> derivativeOfOutWrtInput = new HashMap<>();
+        for (Map.Entry<VertexId, DoubleTensor> entry : derivativeOfOutputsWithRespectToSelf.asMap().entrySet()) {
+            VertexId of = entry.getKey();
+            DoubleTensor partialOfWrtSelf = entry.getValue();
+
+            int[] outputShape = Arrays.copyOf(partialOfWrtSelf.getShape(), partialOfWrtSelf.getShape().length - getShape().length);
+            DoubleTensor ones = DoubleTensor.ones(TensorShape.concat(outputShape, inputVertex.getShape()));
+            DoubleTensor partial = ones.times(partialOfWrtSelf);
```

Comment:
I've seen this trick of making `ones` of a desired shape to stretch the lower rank tensor. Thoughts on pulling it out into a method? I think it'll make it clearer what's going on. 

```
    @Override
    public DoubleTensor stretch(int[] stretchedShape) {
        //check it's legal stretch shape
        DoubleTensor ones = DoubleTensor.ones(stretchedShape);
        return ones.times(this);
    }
```

Reply:
I don't like giving it a name that's not really well defined in the tensor op world. We could go with a broadcastMultiplyFromLeft and right and that would need to be done with all ops that multiply. That'd be a lot of code and at the moment this only happens in 3 places. I'd say lets revisit it in another PR if it comes up more.